core/src/main/java/io/paradoxical/cassieq/auth/SignedRequestAuthFilter.java

```diff
@@ -0,0 +1,171 @@
+package io.paradoxical.cassieq.auth;
+
+import com.godaddy.logging.Logger;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Optional;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import io.dropwizard.auth.AuthFilter;
+import io.dropwizard.auth.AuthenticationException;
+import io.paradoxical.cassieq.model.accounts.AccountName;
+import io.paradoxical.cassieq.model.auth.AccountKeyParameters;
+import io.paradoxical.cassieq.model.auth.AuthorizationLevel;
+import io.paradoxical.cassieq.model.auth.AuthorizedRequestCredentials;
+import io.paradoxical.cassieq.model.auth.RequestParameters;
+import io.paradoxical.cassieq.model.auth.SignedRequestParameters;
+import io.paradoxical.cassieq.model.auth.SignedUrlParameters;
+import lombok.Builder;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Priority;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.IOException;
+import java.security.Principal;
+import java.util.EnumSet;
+import java.util.List;
+
+import static com.godaddy.logging.LoggerFactory.getLogger;
+
+@Priority(Priorities.AUTHENTICATION)
+@Builder(builderClassName = "Builder")
+public class SignedRequestAuthFilter<P extends Principal> extends AuthFilter<AuthorizedRequestCredentials, P> {
+    private static final Logger logger = getLogger(SignedRequestAuthFilter.class);
+
+    private final String accountNamePathParameter;
+    private final String keyAuthPrefix;
+
+    public SignedRequestAuthFilter(String accountNamePathParameter, final String keyAuthPrefix) {
+        this.accountNamePathParameter = accountNamePathParameter;
+        this.keyAuthPrefix = keyAuthPrefix;
+    }
+
+    @Override
+    public void filter(final ContainerRequestContext requestContext) throws IOException {
+        try {
+
+            final MultivaluedMap<String, String> pathParameters = requestContext.getUriInfo().getPathParameters();
+            if (pathParameters.containsKey(accountNamePathParameter)) {
+
+                final AccountName accountName = AccountName.valueOf(pathParameters.getFirst(accountNamePathParameter));
+
+                final RequestParameters headerRequestParameters = parseHeaderRequestParameters(requestContext, accountName);
+                final SignedUrlParameters signedUrlParameters = parseSignedUrlParameters(requestContext, accountName);
+
+                final AuthorizedRequestCredentials credentials =
+                        AuthorizedRequestCredentials.builder()
+                                                    .requestParameters(MoreObjects.firstNonNull(headerRequestParameters, signedUrlParameters))
```

Comment:
Also cool that we can take precedence here of header based auth vs query param based auth